.travis.yml

```diff
@@ -5,16 +5,28 @@ go_import_path: github.com/pingcap/tidb
 go:
   - "1.11.3"
 
-env:
-    - TRAVIS_COVERAGE=0
-    - TRAVIS_COVERAGE=1
+cache:
+  directories:
+    - $GOPATH/pkg/mod
+  # TODO: If memory permits, we could cache this dirctory.
+  # - $HOME/.cache/go-build
+
+env: 
+    - TRAVIS_COVERAGE=0 TEST_SUITE=check
+    - TRAVIS_COVERAGE=0 TEST_SUITE=checkdep
+    - TRAVIS_COVERAGE=0 TEST_SUITE=gotest
+    - TRAVIS_COVERAGE=0 TEST_SUITE=explaintest
+    - TRAVIS_COVERAGE=1 TEST_SUITE="dev upload-coverage"
 
-# Run coverage tests.
 matrix:
+  include:
+    - go: "1.12"
```

Comment:
Now, we could add a newer environment to build and test it.
Take it easy, if this build failure, it will not take effect in CI. 
It only for you to check it manually.

Reply:
F.Y.R: https://docs.travis-ci.com/user/build-matrix/#rows-that-are-allowed-to-fail